dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -288,154 +278,408 @@ def read_metadata(
+                    # There doesn't need to be a file_path
+                    # in all cases for Fastparquet
+                    fpath = pf.fn
+                    base_path = base_path or ""
                 else:
-                    i_path = 0
-                    path_last = path_curr
-                file_path = base_path + path_curr
+                    raise ValueError(
+                        "Global metadata structure is missing a file_path string. "
+                        "If the dataset includes a _metadata file, that file may "
+                        "have one or more missing file_path fields."
+                    )
+
+            if file_row_groups[fpath]:
+                file_row_groups[fpath].append(file_row_groups[fpath][-1] + 1)
             else:
-                # Case (3C)
-                # We cannot pass a specific file/part path
-                file_path = paths
-                i_path = i
-
-            # Strip down pf object
-            if pf_deps and pf_deps != "tuple":
-                # Case (2)
-                for col in part.columns:
-                    col.meta_data.statistics = None
-                    col.meta_data.encoding_stats = None
-
-            # Final definition of "piece" and "pf" for this output partition
-            piece = i_path if pf_deps else part
-            pf_piece = (file_path, gather_statistics) if pf_deps == "tuple" else pf_deps
-            part_item = {
-                "piece": piece,
-                "kwargs": {"pf": pf_piece},
-            }
-            parts.append(part_item)
+                file_row_groups[fpath].append(0)
+
+            # Store mapping of local row-group to global row_group
+            global_lookup[(fpath, file_row_groups[fpath][-1])] = rg
+
+            if gather_statistics:
+                if single_rg_parts:
+                    s = {
+                        "file_path_0": fpath,
+                        "num-rows": row_group.num_rows,
+                        "total_byte_size": row_group.total_byte_size,
+                        "columns": [],
+                    }
+                else:
+                    s = {
+                        "num-rows": row_group.num_rows,
+                        "total_byte_size": row_group.total_byte_size,
+                    }
+                cstats = []
+                for name, i in stat_col_indices.items():
+                    column = row_group.columns[i]
+                    col = column.meta_data.path_in_schema[0]
+                    if column.meta_data.statistics:
+                        cmin = None
+                        cmax = None
+                        if pf.statistics["min"][col][0] is not None:
+                            cmin = pf.statistics["min"][col][rg]
+                            cmax = pf.statistics["max"][col][rg]
+                        elif dtypes[col] == "object":
+                            cmin = column.meta_data.statistics.min_value
```

Comment:
min/max_value is not guaranteed for older datasets, but min/max are fallbacks (the difference is about whether the sorting order on a column is marked - in my experience, never)

Reply:
So, are you suggesting we try to use *_value, but fall back onto min/max if it is None?

Reply:
Yes, I think that might be necessary; this would not be picked up by our tests.